packages/buffetjs-core/src/components/DatePicker/index.js

```diff
@@ -35,107 +51,105 @@ class DatePicker extends React.PureComponent {
-    this.setState({
-      visible: false,
-    });
+  const handleOutsideClick = () => {
+    toggleDatepicker(false);
   };
 
-  handleTabClick = ({ keyCode, which }) => {
+  const handleTabClick = ({ keyCode, which }) => {
     const code = keyCode || which;
 
     if (code === 9) {
-      this.handleOutsideClick();
+      handleOutsideClick();
     }
   };
 
-  showDatepicker = () => {
-    this.setState({
-      visible: true,
-    });
+  const toggleDatepicker = shown => {
+    setState(prevState => ({
+      ...prevState,
+      visible: shown,
+      isFocused: shown,
+    }));
   };
 
-  render() {
-    const {
-      className,
-      disabled,
-      displayFormat,
-      id,
-      name,
-      readOnly,
-      tabIndex,
-    } = this.props;
-    const { date, displayedDate, isFocused, visible } = this.state;
-    let dateValue = date ? moment(date).format(displayFormat) : '';
-
-    if (visible) {
-      dateValue = displayedDate;
-    }
-
-    return (
-      <StyledDatepicker className={`${className} ${visible && 'is-open'}`}>
-        <div>
-          <Input
-            disabled={disabled}
-            type="text"
-            name="start_date"
-            id={id || name}
-            value={dateValue}
-            readOnly={readOnly}
-            onChange={this.handleChange}
-            icon={<FontAwesomeIcon icon={faCalendarAlt} />}
-            onClick={this.showDatepicker}
-            onFocus={this.showDatepicker}
-            onKeyDown={this.handleTabClick}
-            tabIndex={tabIndex}
-          />
-        </div>
-        {visible && (
-          <DayPickerSingleDateController
-            date={date}
-            focused={isFocused}
-            numberOfMonths={1}
-            keepOpenOnDateSelect
-            onFocusChange={this.handleFocusChange}
-            onDateChange={this.handleDateChange}
-            onOutsideClick={this.handleOutsideClick}
-          />
-        )}
-      </StyledDatepicker>
-    );
-  }
+  return (
+    <StyledDatepicker className={`${className} ${state.visible && 'is-open'}`}>
+      <div>
+        <Input
+          disabled={disabled}
+          type="text"
+          name="start_date"
+          id={id || name}
+          value={getDateValue()}
+          readOnly={readOnly}
+          onChange={handleChange}
+          icon={<FontAwesomeIcon icon={faCalendarAlt} />}
+          onClick={() => toggleDatepicker(true)}
+          onKeyDown={handleTabClick}
+          tabIndex={tabIndex}
+        />
+      </div>
+      {state.isFocused && (
+        <DayPickerSingleDateController
+          date={state.date}
+          focused
+          numberOfMonths={1}
+          onDateChange={handleDateClick}
+          onOutsideClick={handleOutsideClick}
+          daySize={37}
+          transitionDuration={0}
+        />
+      )}
+    </StyledDatepicker>
+  );
 }
 
-DatePicker.defaultProps = {
+Datepicker.defaultProps = {
   className: null,
   disabled: false,
   displayFormat: 'MMMM DD, YY',
```

Comment:
```suggestion
  displayFormat: 'MMMM DD, YYYY',
```

Generally a better standard